asf_floorplan_interpreter/pipeline/annotation/floorplan_recipe.py

```diff
@@ -0,0 +1,81 @@
+import prodigy
+from prodigy.components.loaders import JSONL
+from ultralytics import YOLO
+
+
+def load_model(directory):
+    """Function to load YOLO model trained on floor plans annotated with polygons identifying rooms"""
+    return YOLO(directory)
+
+
+def yolo_2_segments(results):
+    segments = []
+    for (x, y, w, h), label in zip(results[0].boxes.xywh, results[0].boxes.cls):
+        x_min = x.item() - (w.item() / 2)
+        y_min = y.item() - (h.item() / 2)
+        x_max = x.item() + (w.item() / 2)
+        y_max = y.item() + (h.item() / 2)
+        segment = [[x_min, y_min], [x_max, y_min], [x_max, y_max], [x_min, y_max]]
+        segments.append(
+            {
+                "label": results[0].names[label.item()],
+                "points": segment,
+                "type": "polygon",
+            }
+        )
+    return segments
```

Comment:
fyi @cmbrennan002 - this is where I convert the output of Adeola's model to points for Prodgiy

Reply:
Thank you! :)